### PR TITLE
feat(providers): add netlify

### LINKED ISSUE #714: Provider request: Netlify
Netlify offers managed DNS for arbitrary zones.

There is an API, although the docs and Go client support are pending in a PR: netlify/open-api#207

### PR DESCRIPTION
**Pull request by SphericalKat**
Closes #714

### CODE REVIEW

**Comment by tlimoncelli** [comment]
Looks good so far!

Just FYI: It is normal for tests to fail as "unauthorized".  Github won't provide API keys to PRs that come from outside.  

Let me know when its ready for for review.

**Comment by SphericalKat** [comment]
Hello, this is ready for review. Thanks!

**Comment by msfjarvis** [comment]
Found a couple issues while testing this out

1. Netlify's `NETLIFY` and `NETLIFYv6` records are not supported but definitely should be
2. TXT records are currently broken, but that's a quick fix

```diff
diff --git providers/netlify/netlifyProvider.go providers/netlify/netlifyProvider.go
index 2752274fd229..4bdfb6580965 100644
--- providers/netlify/netlifyProvider.go
+++ providers/netlify/netlifyProvider.go
@@ -116,7 +116,7 @@ func (n *netlifyProvider) GetZoneRecords(domain string) (models.Records, error)
 			}
 			err = rec.SetTargetSRV(uint16(r.Priority), r.Weight, r.Port, r.Value)
 		case "TXT":
-			err = rec.SetTargetTXTfromRFC1035Quoted(r.Value)
+			err = rec.SetTargetTXT(r.Value)
 		case "CAA":
 			err = rec.SetTargetCAA(uint8(r.Flag), r.Tag, r.Value)
 		default:
```

**Comment by SphericalKat** [comment]
Hello, I've been working with @msfjarvis to fix the issues he reported. The TXT records issue has been fixed, but NETLIFY records seem to be rather painful. Through a bit of experimentation, it's clear that the Netlify API does not allow for manual creation of these records. 

One option is just to transparently ignore them, but I'm open to other suggestions.

cc: @tlimoncelli

**Comment by tlimoncelli** [comment]
I concur.  Let's break this into 2 merges.  The first will ignore NETLIFY records.  Then work out how to handle them in a second merge.

**Comment by abovedave** [comment]
It seems like `NETLIFY` records are special in the sense they are created/updated at the site level within the Netlify UI (for features like [branch subdomains](https://docs.netlify.com/domains-https/custom-domains/multiple-domains/#branch-subdomains)).

Perhaps it's worth exploring the [`updateSite`](https://open-api.netlify.com/#tag/site/operation/updateSite) endpoint?

**Comment by msfjarvis** [comment]
Likely unsuitable to be merged into the PR, but with [this diff](https://msfjarvis.dev/g/dnscontrol/c93a532f50d1) applied I've been able to use `IGNORE_NAME` to suppress the unwanted deletion of the `NETLIFY` records.

```javascript
D('msfjarvis.dev', NewRegistrar('none'), DnsProvider(NewDnsProvider('netlify')),
  IGNORE_NAME('*', 'NETLIFY'),
  IGNORE_NAME('*', 'NETLIFYv6'),
);
```

**Comment by abovedave** [comment]
Been thinking about this some more, as `NETLIFY` records are handles automatically when you create or delete sites, push branches etc, is there even as use case for modifying them through DNScontrol?

I'm think the solution might be just transparently ignore as suggested.

File: pkg/js/helpers.js [hunk 1]
```diff
@@ -410,6 +410,9 @@ var NAPTR = recordBuilder('NAPTR', {
     },
 });
 
+var NETLIFY = recordBuilder('NETLIFY')
+var NETLIFYv6 = recordBuilder('NETLIFYv6')
```
Comment (tlimoncelli) [inline pkg/js/helpers.js:414]: Are NETLIFY and NETLIFYv6 needed if we're going to ignore them?

**Review by tlimoncelli — APPROVED** [review]
As far as the code is concerned, this all looks great!  No suggestions.

I won't merge it until I hear from you about ignoring the NETLIFY records, etc.

**Comment by SphericalKat** [comment]
Hey, just got back to work on this. No, we don't need NETLIFY stuff in the helpers; I'll remove those and modify the provider to ignore NETLIFY records in a moment.

File: pkg/js/helpers.js (hunk 1 above)
Comment (SphericalKat) [reply pkg/js/helpers.js:414]: Resolved in dc10565c

**Review by msfjarvis — APPROVED** [review]
Verified that the provider correctly skips the NETLIFY and NETLIFYv6 rtypes for my site

**Comment by tlimoncelli** [comment]
Thank you for the contribution!  A lot of people have asked for this!

**Comment by SphericalKat** [comment]
Cheers!

**Comment by j-f1** [comment]
Thanks for adding this! Can you add a clarification to the docs about why someone would want to provide (or not provide) an account slug?

**Comment by j-f1** [comment]
@SphericalKat Not sure if you saw my comment above but it was a minor point of confusion as I was setting up my Netlify configuration ^

**Pull request merged**